File: somdutta/solution/naive.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
long fact(long n)
{
      if(n==0)
      return 1;
    long res = 1;
    for (long i = 2; i <= n; i++)
        res = res * i;
    return res;
}

long nCr(long n, long r)
{
    return fact(n) / (fact(r) * fact(n - r));
}

// Returns factorial of n

int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int t;
    cin>>t;
    while(t--){
        int x, y;
        cin>>x>>y;
        unsigned long long int sum = 0;
        for(long i=0;i<=x;i++){
            sum += long(pow(y, x-i))*nCr(x, i);
        }
        cout<<sum<<endl;
    }
    return 0;
}

File: somdutta/solution/naive.py
def nCr(n, r):

    return (fact(n) / (fact(r) * fact(n - r)))

# Returns factorial of n
def fact(n):
    if n == 0 or n==1:
        return 1
    res = 1
    
    for i in range(2, n+1):
        res = res * i
        
    return res

t = int(input())
while t > 0:
    user_input = input()

# Split the input string into two parts
    x, y = map(int, user_input.split())
    sum = 0
    for i in range(x+1):
        sum += pow(y, x-i)*nCr(x, i)
    print(int(sum))
    t = t - 1